### PR TITLE
Manually reorder event types

### PR DESCRIPTION
**Pull request by pumfleet**
This PR adds a new `position` field to event types, allowing the user to reorder their event types from within the UI.

https://www.loom.com/share/fa927865d7fd4a53a847ef5f45dec820

### CODE REVIEW

File: pages/event-types/index.tsx [hunk 1]
```diff
@@ -1,8 +1,15 @@
 // TODO: replace headlessui with radix-ui
 import { Menu, Transition } from "@headlessui/react";
-import { UsersIcon } from "@heroicons/react/solid";
-import { ChevronDownIcon, PlusIcon } from "@heroicons/react/solid";
-import { DotsHorizontalIcon, ExternalLinkIcon, LinkIcon } from "@heroicons/react/solid";
+import {
```
Comment (baileypumfleet) [inline pages/event-types/index.tsx:3]: I added two icons and merged the above 3 lines into a single import.

File: public/static/locales/en/common.json [hunk 2]
```diff
@@ -511,5 +511,6 @@
   "delete_event_type": "Delete Event Type",
   "confirm_delete_event_type": "Yes, delete event type",
   "integrations": "Integrations",
-  "settings": "Settings"
+  "settings": "Settings",
```
Comment (baileypumfleet) [inline public/static/locales/en/common.json:514]: I hope that this is all I need to add a new string. If there's something I'm missing, please let me know.

**Comment by KATT** [comment]
I can check this out on Monday! It looks good on first glance, main thing is that you likely need to send _all_ event types order and not just one - - Imagine 3 event types have order 0 and increment two of them, then there'd be two event types with the same ordering.

Secondly, the ordering should probably be DESC on order and secondary order on date created ASC so new ones are always in bottom until an order index is applied.

Thirdly, the mutation need optimistic updates and/or two invalidate the cache once the mutation is applied (utils.invalidateQueries(..))


On my phone, will do more helpful comments on Monday!

**Comment by PeerRich** [comment]
i like "latest event-type at the top" ordering

**Comment by KATT** [comment]
> i like "latest event-type at the top" ordering

As a user it's not what I'd expect to happen, but take it to a vote? 😂

**Comment by PeerRich** [comment]
> > i like "latest event-type at the top" ordering
> 
> As a user it's not what I'd expect to happen, but take it to a vote? 😂

sure i don't mind voting haha. I am open to whatever order to be fair

File: server/routers/viewer.tsx [hunk 3]
```diff
@@ -423,6 +449,106 @@ const loggedInViewerRouter = createProtectedRouter()
...
+      await Promise.all(
+        _.reverse(input.ids).map((id, position) => {
+          console.log({
+            where: {
+              id,
+            },
+            data: {
+              position,
+            },
+          });
```
Comment (KATT) [inline server/routers/viewer.tsx:505]: was there for debugging
```suggestion
```

File: server/routers/viewer.tsx [hunk 4]
```diff
@@ -135,6 +145,14 @@ const loggedInViewerRouter = createProtectedRouter()
...
             },
             select: eventTypeSelect,
+            orderBy: [
+              {
+                position: "desc",
+              },
+              {
+                id: "asc",
+              },
+            ],
```
Comment (KATT) [inline server/routers/viewer.tsx:155]: a bit ugly to have this repeated - would be nicer to have it as function or a way to set default sorting in prisma

**Comment by baileypumfleet** [comment]
Order by length by default?

**Comment by KATT** [comment]
> Order by length by default?

That will feel random to the users when they add new events unless it has radio options where you could select "(x) order by length / ( ) custom order"

**Comment by FlxMgdnz** [comment]
> Order by length by default?

I'd like that very much. Ascending of course.

**Review by KATT — COMMENTED** [review]
LGTM but I also touched it - maybe @zomars / @mihaic195 / @emrysal can review

**Comment by PeerRich** [comment]
what happened to the drag & drop UI that me and @zomars build? Will it be merged?

**Review by PeerRich — APPROVED** [review]

**Comment by KATT** [comment]
> what happened to the drag & drop UI that me and @zomars build? Will it be merged?

re drag & drop --

Screenshot 2021-11-15 at 12 49 16

**Comment by PeerRich** [comment]
> > what happened to the drag & drop UI that me and @zomars build? Will it be merged?
> 
> re drag & drop --
> 
> Screenshot 2021-11-15 at 12 49 16

ah perfect. i am all for follow up PRs 🎉 lets get this merged @baileypumfleet

**Pull request merged**